tests/test_csc.py

```diff
@@ -128,6 +130,117 @@ async def test_simple_follow(self):
             )
             await self.assert_dome_az(azimuth=None, move_expected=False)
 
+    async def test_telescope_vignetted(self):
+        async with self.make_csc(initial_state=salobj.State.ENABLED):
+            angle_margin = 0.01
+            await self.assert_next_sample(self.remote.evt_followingMode, enabled=False)
+            vignette_min = self.csc.config.vignette_min
+            vignette_max = self.csc.config.vignette_max
+
+            assert self.dome_csc.curr_az == 0
+            # Wait for ATDomeTrajectory to notice that both shutter doors
+            # are fully open.
+            while True:
+                data = await self.assert_next_sample(
+                    topic=self.remote.evt_telescopeVignetted,
+                    azimuth=TelescopeVignetted.UNKNOWN,
+                )
+                if data.shutter == TelescopeVignetted.NO:
+                    break
+            await self.publish_telescope_actual_position(azimuth=0, elevation=0)
+            await self.assert_next_sample(
+                topic=self.remote.evt_telescopeVignetted,
+                azimuth=TelescopeVignetted.NO,
+                shutter=TelescopeVignetted.NO,
+            )
+
+            # Move the dome far enough negative to vignette partially
+            dome_az = 0 - vignette_min - angle_margin
+            await self.dome_remote.cmd_moveAzimuth.set_start(azimuth=dome_az)
+            await self.assert_next_sample(
+                topic=self.remote.evt_telescopeVignetted,
+                azimuth=TelescopeVignetted.PARTIALLY,
+                shutter=TelescopeVignetted.NO,
+            )
+
+            # Change telescope azimuth far enough away on the other side
+            # of zero to fully vignette
+            telescope_az = dome_az + vignette_max + angle_margin
+            await self.publish_telescope_actual_position(
+                azimuth=telescope_az, elevation=0
+            )
+            await self.assert_next_sample(
+                topic=self.remote.evt_telescopeVignetted,
+                azimuth=TelescopeVignetted.FULLY,
+                shutter=TelescopeVignetted.NO,
+            )
+
+            # Increase telescope elevation high enough to partial vignetting.
+            # Rather than be fancy about the math, just use a plausible value.
+            await self.publish_telescope_actual_position(
+                azimuth=telescope_az, elevation=45
+            )
+            await self.assert_next_sample(
+                topic=self.remote.evt_telescopeVignetted,
+                azimuth=TelescopeVignetted.PARTIALLY,
+                shutter=TelescopeVignetted.NO,
+            )
+
+            # Increase telescope elevation high enough to full vignetting.
+            # Rather than be fancy about the math, just use a plausible value.
+            await self.publish_telescope_actual_position(
+                azimuth=telescope_az, elevation=90
```

Comment:
As this is alt-az, I will rather avoid using elevation=90..this is an untrackable area, can you change that to elevation=89, please?

Reply:
I'm not convinced. Most az/el telescope are perfectly happy to go to 90 degrees elevation. They just cannot go there while tracking a sidereal object. Vignetting is independent of whether the object is sidereal or not.